app/controllers/TransferAgreementController.scala

```diff
@@ -39,8 +43,29 @@ class TransferAgreementController @Inject()(val controllerComponents: SecurityCo
     )(TransferAgreementData.apply)(TransferAgreementData.unapply)
   )
 
-  def transferAgreement(consignmentId: Long): Action[AnyContent] = secureAction { implicit request: Request[AnyContent] =>
-    Ok(views.html.transferAgreement(consignmentId, transferAgreementForm, options))
+  private val getConsignmentClient = graphqlConfiguration.getClient[getConsignment.Data, getConsignment.Variables]()
+
+  private def getConsignmentDetails(request: Request[AnyContent],
+                                    status: Status,
+                                    consignmentId: Long)(
+                                     implicit requestHeader: RequestHeader): Future[Result] = {
+    val variables: getConsignment.Variables = new GetConsignment.getConsignment.Variables(consignmentId)
+    getConsignmentClient.getResult(request.token.bearerAccessToken, getConsignment.document, Some(variables)).map(data => {
+      if (data.data.isDefined) {
+        val consignmentData: Option[(Long, UUID)] = data.data.get.getConsignment.map(c => (c.seriesid, c.userid))
```

Comment:
All we really care about here is whether the consignment exists or not - we don't actually need the series ID or user ID. It would be better to remove this in case the API changes in the future. For example, we might remove userid from the response because the frontend doesn't normally need to know it.

What's the difference between an API response with consignment data, and one without it? How can we check for that? (That's not a leading question, btw - I can't actually remember what the API returns when there's no data.)

Reply:
I can do that, would you like me to change the query in the generated GQL project to reflect this as well? Something like:
```query getConsignment($consignmentId: Long!) {
  getConsignment(consignmentid: $consignmentId) {
    consignmentid
  }
}